### DISCUSSION TITLE
Change Storage Folder

### DISCUSSION THREAD

**Discussion by JEEVAN28** [YetiForceCompany/YetiForceCRM#14914] [How to]
Hi,
I would like to change Storage Folder for Document from default installation folder to connected shares NAS folder to ensure that all images and documents are properly back up. How do I do this.

Regards
Jeevan Pendhari

**Comment by diderich** [comment]
On Linux, copy the folder `storage` to your NAS. Create a symbolic link to the "NAS" using `ln -s "NAS"/storage storage` in the root directory of your YetiForce installation.

And don't forget to backup the database in parallel, as only backing-up the storage directory does not allow you to restore the system in case of failure.

And make a backup of your YetiForce installation each time files have been changed, i.e., updates made to the system, that is aligned with your backed-up database.

Maybe it would be easier to move the whole YetiForce installation, including the database, to your NAS.

**Comment by drakub** [comment]
Hello, am in this same loop. I have managed to point the upload to a NAS, but when it comes to downloading the file, I get a blank screen. The file is actually uploaded in the specified directory (external NAS) but in the download, it is downloading the file. What can I do? Attached is the screen am getting when I attempt to download.

![image](https://github.com/YetiForceCompany/YetiForceCRM/assets/19745222/1ae5033b-00a6-433c-94a4-74efa3e20726)

**Reply by mariuszkrzaczkowski** [reply]
logs ...

**Comment by drakub** [comment]
No logs related to this but I have observed that the file download looks at
the ROOT_DIRECTORY as base folder before the storagePath value.


[image: image.png]

This means that it will keep looking for the storage location within the
partition/disk where the application is.

As a test drive, I modified this to read the storage location in the
configs and it works perfectly well. But I am pretty sure that is not
a sustainable solution since different people will be managing upgrades to
the system.



*Draku Job*
*Tel: +256 786 610580*
*B. Information Systems*
*Systems Developer | Database Designer |**Technical Writer | ICT Instructor*
